### DISCUSSION TITLE
RTL is not connecting over Tor

### DISCUSSION THREAD

**Discussion by n-e-t-d-i-v-e-r** [Ride-The-Lightning/RTL#1273] [C-Lightning]
Hello
I have installed C-lightning and then also connected RTL to it. I can also access the page locally via localhost as expected and it works fine so far.

Unfortunately, however, I cannot access the page via the Tor browser.

My configuration looks like this:

*cl-rest-config.json:

{
"PORT": 3092,
"DOCPORT": 4092,
"PROTOCOL": "http",
"EXECMODE": "production",
"RPCCOMMANDS": ["*"],
"DOMAIN": "localhost",
}

/data/lightningd/config:

network

proxy=127.0.0.1:9050
bind-addr=0.0.0.0:9736
addr=statictor:127.0.0.1:9051/torport=9736
always-use-proxy=true

cln-rest-plugin

plugin=/data/lightningd-plugins-available/c-lightning-REST-0.10.1/clrest.js
rest-port=3092
rest-docport=4092
rest-protocol=http

Servers:

[ 1] 22/tcp ALLOW IN Anywhere
[ 2] 50002/tcp ALLOW IN Anywhere # allow Electrum SSL
[ 3] 4000/tcp ALLOW IN Anywhere # allow BTC RPC Explorer SSL
[ 4] 4001/tcp ALLOW IN Anywhere # allow RTL SSL
[ 5] 4002 ALLOW IN Anywhere # allow ThunderHub SSL
[ 6] 4081/tcp ALLOW IN Anywhere # allow Mempool SSL
[ 8] 4091/tcp ALLOW IN Anywhere # allow Homer SSL
[ 9] 4003/tcp ALLOW IN Anywhere # allow LNBits SSL

[11] 50002/tcp (v6) ALLOW IN Anywhere (v6) # allow Electrum SSL
[12] 4000/tcp (v6) ALLOW IN Anywhere (v6) # allow BTC RPC Explorer SSL
[13] 4001/tcp (v6) ALLOW IN Anywhere (v6) # allow RTL SSL
[14] 4002 (v6) ALLOW IN Anywhere (v6) # allow ThunderHub SSL
[15] 4081/tcp (v6) ALLOW IN Anywhere (v6) # allow Mempool SSL
[16] 4091/tcp (v6) ALLOW IN Anywhere (v6) # allow Homer SSL
[17] 4003/tcp (v6) ALLOW IN Anywhere (v6) # allow LNBits SSL

TOR hidden-services:

Hidden service LNBits

HiddenServiceDir /var/lib/tor/hidden_service_lnbits/
HiddenServiceVersion 3
HiddenServicePort 80 127.0.0.1:5000

Hidden Service Thunderhub_LND

HiddenServiceDir /var/lib/tor/hidden_service_thunderhub/
HiddenServiceVersion 3
HiddenServicePort 80 127.0.0.1:3010

#RTL_CLN Hidden Service
HiddenServiceDir /var/lib/tor/hidden_service_cln_rest/
HiddenServiceVersion 3
HiddenServicePort 8080 127.0.0.1:3092

RTL-config.json:

{
"port": "9090",
"SSO": {
"rtlSSO": 0,
"rtlCookiePath": "",
"logoutRedirectLink": ""
},
"nodes": [
{
"index": 1,
"lnNode": "CLN Node",
"lnImplementation": "CLN",
"Authentication": {
"macaroonPath": "/home/rtl",
"configPath": "/data/lightningd/config"
},
"Settings": {
"userPersona": "OPERATOR",
"themeMode": "DAY",
"themeColor": "INDIGO",
"fiatConversion": true,
"currencyUnit": "EUR",
"logLevel": "ERROR",
"lnServerUrl": "http://127.0.0.1:3092",
"enableOffers": true
}
}
],

Anybody has got an idea, where the issue in this config is?
<img width="1029" alt="Screenshot 2023-08-01 at 09 32 12" src="https://github.com/Ride-The-Lightning/RTL/assets/87816768/a1030c32-0368-4a2c-8527-39c2ffb823b7">

**Comment by saubyk** [comment]
Hi @n-e-t-d-i-v-e-r you need to configure RTL under the Tor hidden service, not c-lightning-rest.

**Comment by n-e-t-d-i-v-e-r** [comment]
I’m not sure understand you right, but I did cofigure RTL under the Tor hidden service, as you can see here. But it dosen’t works:

TOR hidden-services:

Hidden service LNBits

HiddenServiceDir /var/lib/tor/hidden_service_lnbits/
HiddenServiceVersion 3
HiddenServicePort 80 127.0.0.1:5000

Hidden Service Thunderhub_LND

HiddenServiceDir /var/lib/tor/hidden_service_thunderhub/
HiddenServiceVersion 3
HiddenServicePort 80 127.0.0.1:3010

#RTL_CLN Hidden Service
HiddenServiceDir /var/lib/tor/hidden_service_cln_rest/
HiddenServiceVersion 3
HiddenServicePort 8080 127.0.0.1:3092

**Reply by saubyk** [reply]
cln_rest is the REST API server for the RTL backend. You need to configure RTL under Tor to access RTL not it's api server

**Reply by saubyk** [reply]
So it should be something like this instead...
```
#RTL_CLN Hidden Service
HiddenServiceDir /var/lib/tor/hidden_service_rtl/
HiddenServiceVersion 3
HiddenServicePort 80 127.0.0.1:9090
```

**Comment by n-e-t-d-i-v-e-r** [comment]
> cln_rest is the REST API server for the RTL backend. You need to configure RTL under Tor to access RTL not it's api server

**Comment by n-e-t-d-i-v-e-r** [comment]
That was the issue! Thx for support. I copied exactly the conig you pasted into tor-config and now RTL appears in tor-browser.
